contrib/remote-ssh-build.bash

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# vim: softtabstop=2 shiftwidth=2 expandtab
+
+## REMOTE SSH ACCESS
+
+# This script builds a zbm with remote access support via ssh. The image and
+# efi executable will contain private host keys and public keys used for
+# mutual authentication.
+
+# All keys / settings are placed into ${RS_DIR} ('ssh-data' in the current
+# working directory by default). Its contents can be customized as needed.
+# They are regenerated if missing.
+
+# Server private keys are generated automatically as needed. It is
+# recommended to not reuse existing private keys.
+# User authentication for the root account is done via 'authorized_keys'.
+# The current users' authorized keys are copied if it is not present.
+
+RS_DIR=$(realpath "${RS_DIR:-ssh-data}")
+RS_KEYDIR="${RS_DIR}/keys"
+
+mkdir -p "${RS_DIR}"
+mkdir -p "${RS_KEYDIR}"
+
+
+## PREPARE SERVER PRIVATE KEYS
+
+# If no local ssh host keys are available, create them
+# WARNING: These private keys will be part of the final image!
+RS_PK_RSA="${RS_KEYDIR}"/ssh_host_rsa_key
+RS_PK_ECDSA="${RS_KEYDIR}"/ssh_host_ecdsa_key
+RS_KEYGEN_OPT=(-m PEM -N '' -C "zbm-server-key")
+if [ ! -f "${RS_PK_RSA}" ]; then
+  ssh-keygen -t rsa "${RS_KEYGEN_OPT[@]}" -f "${RS_PK_RSA}"
+fi
+if [ ! -f "${RS_PK_ECDSA}" ]; then
+  ssh-keygen -t ecdsa "${RS_KEYGEN_OPT[@]}" -f "${RS_PK_ECDSA}"
+fi
```

Comment:
Make this more general to allow easty and arbitrary customization of the keytypes to be generated.
```suggestion
RS_SSH_KEYTYPES=(rsa ecdsa)
RS_KEYGEN_OPT=(-m PEM -N '' -C "zbm-server-key")
for keytype in "${RS_SSH_KEYTYPES[@]}"; do
  RS_PK_FILE="${RS_KEYDIR}/ssh_host_${keytype}_key"
  [ -f "${RS_PK_FILE}" ] && continue
  ssh-keygen -t "${keytype}" "${RS_KEYGEN_OPT[@]}" -f "${RS_PK_FILE}"
done
```

Reply:
I like this improvement, but I don't think any other key formats will actually be useful (not without also changing the config file for dropbear in sync, and I'm not sure if it supports more formats)

Reply:
We support three key types in [dracut-crypt-ssh](https://github.com/dracut-crypt-ssh/dracut-crypt-ssh/commit/6ae8c358d24989ac7c6e625609bd1af4cf8ce1de); it is probably equally useful to allow people to strip out unwanted keys as it is to allow them to add something like ed25519.

Reply:
I've added this functionality, but somehow dropbear doesn't like ed25519. Maybe in the future.